### ISSUE TITLE
"optional" does not exist on computed optional class property with variable

### ISSUE THREAD

**Issue by sosukesuzuki** [typescript-eslint/typescript-eslint#1019]
(This issue related on https://github.com/prettier/prettier/issues/6569)

**What code were you trying to parse?**

```ts
// Put your code here
const bar = "bar";
class Foo {
    [bar]?;
}
```

**What did you expect to happen?**
```js
{
  "type": "ClassProperty",
  "key": {
    "type": "Identifier",
    "name": "bar",
    "range": [36, 39],
    "loc": {
      "start": {
        "line": 4,
        "column": 5
      },
      "end": {
        "line": 4,
        "column": 8
      }
    }
  },
  "value": null,
  "computed": true,
  "static": false,
  "range": [35, 42],
  "loc": {
    "start": {
      "line": 4,
      "column": 4
    },
    "end": {
      "line": 4,
      "column": 11
    }
  },
  "optional": true // This "optional" should exist
}
```

**What actually happened?**

```js
{
  "type": "ClassProperty",
  "key": {
    "type": "Identifier",
    "name": "bar",
    "range": [36, 39],
    "loc": {
      "start": {
        "line": 4,
        "column": 5
      },
      "end": {
        "line": 4,
        "column": 8
      }
    }
  },
  "value": null,
  "computed": true,
  "static": false,
  "range": [35, 42],
  "loc": {
    "start": {
      "line": 4,
      "column": 4
    },
    "end": {
      "line": 4,
      "column": 11
    }
  }
}
```

It works fine for computed optional properties that do not use variables.

```ts
class Foo {
  ["bar"]?;
  [1]?;
}
```

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `2.3.1` |
| `TypeScript`                           | `3.4.1` |
| `node`                                 | `12.10.0` |
| `npm`                                  | `6.11.3` |

**Comment by bradzacher** [comment]
who the heck is writing and wanting to use such an abomination?!?
What's the point of typescript if you're providing no types....

It's a simple enough fix.

**Comment by bradzacher** [comment]
The fix will be released (automatically) some time this afternoon
